### PR TITLE
[WIP] Upgrade checkerframework dependency to v2.2.0

### PR DESCRIPTION
**Pull request by lazaroclapp**
+ Remove our earlier workaround for lambdas dataflow, now that upstream fix is up (see. https://github.com/typetools/checker-framework/pull/1483 )

This is a WIP right now and will not/should not pass C.I. Putting the PR here to track this.

### CODE REVIEW

**Comment by lazaroclapp** [comment]
Note, this fails with the following stacktrace:

```
com.google.common.util.concurrent.UncheckedExecutionException: java.lang.NullPointerException
  	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2217)
  	at com.google.common.cache.LocalCache.get(LocalCache.java:4154)
  	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:4158)
  	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:5147)
  	at com.google.common.cache.LocalCache$LocalLoadingCache.getUnchecked(LocalCache.java:5153)
  	at com.uber.nullaway.dataflow.DataFlow.methodDataflow(DataFlow.java:122)
```

This is likely NullAway still pulling the old checkerframework classes from the EP fat jar. Will double check this tomorrow. @msridhar , if that's the case, is there still a good workaround that doesn't require us to maintain an EP fork?

**Comment by msridhar** [comment]
So I got things working for the unit tests by only putting JDK 9 javac in the bootclasspath.  But when I tried an "integration" test by sticking the lambda code in the sample project, it still failed.  I fixed that problem by "forcing" the checker dataflow version in `sample/build.gradle`.  Let's think more and do more sanity checking before actually merging this back.

(FYI @lazaroclapp I rebased this PR to get some other `build.gradle` changes)

File: sample/build.gradle [hunk 1]
```diff
@@ -22,6 +22,7 @@ plugins {
 
 configurations.errorprone {
   resolutionStrategy.force "com.google.errorprone:error_prone_core:2.1.1"
+  resolutionStrategy.force "org.checkerframework:dataflow:2.2.0"
```
Comment (msridhar) [inline sample/build.gradle]: @kageiit is it reasonable to ask our users to "force upgrade" Checker Dataflow in this way?  Might they run into problems on other build systems?

File: nullaway/build.gradle [hunk 2]
```diff
@@ -45,7 +45,7 @@ dependencies {
     testCompile deps.apt.javaxInject
     testCompile deps.test.rxjava2
 
-    epCore deps.build.errorProneCore
+    epJavac "com.google.errorprone:javac:9-dev-r4023-3"
```
Comment (msridhar) [inline nullaway/build.gradle]: @kageiit what I really want to here is reference the version of `com.google.errorprone:javac` that `deps.build.errorProneCore` is using.  Is there any way to do that?

Comment (kageiit) [reply nullaway/build.gradle]: Yes. There is a way to extract this version. I can fix it up in a follow up or just push to the same branch. lemme know

File: sample/build.gradle (hunk 1 above)
Comment (kageiit) [reply sample/build.gradle]: If it is just in the `errorprone` configuration, it is quite reasonable till we get the fix upstream

but using `force` is a bad idea as if users forget to unset it, the version will be downgraded on future updates

They should just add the dataflow dependency to the `errorprone` configuration and let regular dependency version resolution take place

Comment (msridhar) [reply sample/build.gradle]: How do we do this?  Feel free to push a change on the branch

File: nullaway/build.gradle (hunk 2 above)
Comment (msridhar) [reply nullaway/build.gradle]: If you could push on the branch, that'd be great, thanks

**Pull request merged**